hyperspy/component.py

```diff
@@ -756,6 +756,9 @@ def summary(self):
                                                parameter.std,
                                                parameter.units)
 
+    def __tempcall2d__(self, p, x, y, onlyfree=True):
+        self.fetch_values_from_array(p, onlyfree=onlyfree)
+        return self.function(x, y)
```

Comment:
I'm convinced: I agree that readability and usability trumps the code reuse in this case. :)